stress/common/src/main/java/alluxio/stress/worker/IOTaskSummary.java

```diff
@@ -0,0 +1,360 @@
+      totalDuration += p.mDuration;
+      totalSize += p.mDataSizeMB;
+      double speed = p.mDataSizeMB / p.mDuration;
+      maxSpeed = Math.max(maxSpeed, speed);
+      minSpeed = Math.min(minSpeed, speed);
+      speeds[i++] = p.mDataSizeMB / p.mDuration;
+    }
+    double avgSpeed = totalSize / totalDuration;
+    double var = 0;
+    for (double s : speeds) {
+      var += (s - avgSpeed) * (s - avgSpeed);
+    }
+
+    result.mTotalDuration = totalDuration;
+    result.mTotalSize = totalSize;
+    result.mMaxSpeed = maxSpeed;
+    result.mMinSpeed = Double.compare(minSpeed, Double.MAX_VALUE) == 0 ? 0.0 : minSpeed;
+    result.mAvgSpeed = avgSpeed;
+    result.mStdDev = Math.sqrt(var);
+
+    return result;
+  }
+
+  private void calculateStats() {
+    List<IOTaskResult.Point> readPoints = mPoints.stream().filter((p) ->
+            p.mMode == IOConfig.IOMode.READ && p.mDuration > 0)
+            .collect(Collectors.toList());
+    mReadSpeedStat = calculateStat(readPoints);
+
+    List<IOTaskResult.Point> writePoints = mPoints.stream().filter((p) ->
+            p.mMode == IOConfig.IOMode.WRITE && p.mDuration > 0)
+            .collect(Collectors.toList());
+    mWriteSpeedStat = calculateStat(writePoints);
+  }
+
+  @Override
+  public GraphGenerator graphGenerator() {
+    return new GraphGenerator();
+  }
+
+  @Override
+  public String toString() {
+    return String.format("IOTaskSummary: {Points=%s, Errors=%s}%n",
+            mPoints, mErrors);
+  }
+
+  /**
+   * A graph generator for the statistics collected.
+   * */
+  public static final class GraphGenerator extends alluxio.stress.GraphGenerator {
+    @Override
+    public List<Graph> generate(List<? extends Summary> results) {
+      List<Graph> graphs = new ArrayList<>();
+      // only examine MaxThroughputSummary
+      List<IOTaskSummary> summaries =
+              results.stream().map(x -> (IOTaskSummary) x).collect(Collectors.toList());
+
+      if (summaries.isEmpty()) {
+        LOG.info("No summaries");
+        return graphs;
+      }
+
+      // TODO(jiacheng): how to include params in this?
+      // first() is the list of common field names, second() is the list of unique field names
+      Pair<List<String>, List<String>> fieldNames = Parameters.partitionFieldNames(
+              summaries.stream().map(x -> x.mParameters).collect(Collectors.toList()));
+      System.out.format("FieldNames: %s%n", fieldNames);
+
+      // Split up common description into 100 character chunks, for the sub title
+      List<String> subTitle = new ArrayList<>(Splitter.fixedLength(100).splitToList(
+              summaries.get(0).mParameters.getDescription(fieldNames.getFirst())));
+      System.out.format("subtitle: %s%n", subTitle);
+
+      for (IOTaskSummary summary : summaries) {
+        String series = summary.mParameters.getDescription(fieldNames.getSecond());
+        subTitle.add(series);
+      }
+
+      BarGraph speedGraph = new BarGraph("Read/Write speed",
```

Comment:
are these any new system requirements for plotting the graphs? or do we include everything we need in the server tarball

Reply:
There's no new system requirements for plotting. The HTML generation in `alluxio.stress.graph.LineGraph/BarGraph` wraps its own JS libraries.

Reply:
The dependencies are wrapped in the `LineGraph` and `BarGraph` classes. They generate their own HTML referring to the JS libs.